proto/elys/amm/tx.proto

```diff
@@ -46,22 +47,24 @@ message MsgExitPool {
 message MsgExitPoolResponse {}
 
 message MsgSwapExactAmountIn {
-           string                   creator           = 1;
-           cosmos.base.v1beta1.Coin tokenIn           = 2 [(gogoproto.nullable)   = false                                    ] ;
-           string                   tokenOutMinAmount = 3 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Uint", (gogoproto.moretags) = "yaml:\"taken_out_min_amount\"", (gogoproto.nullable) = false];
-  repeated uint64                   swapRoutePoolIds  = 4;
-  repeated string                   swapRouteDenoms   = 5;
+           string                   sender           = 1;
+           repeated SwapAmountInRoute routes = 2 [ (gogoproto.nullable) = false ];
+           cosmos.base.v1beta1.Coin tokenIn           = 3 [(gogoproto.nullable)   = false                                    ] ;
+           string                   tokenOutMinAmount = 4 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", (gogoproto.moretags) = "yaml:\"taken_out_min_amount\"", (gogoproto.nullable) = false];
 }
 
-message MsgSwapExactAmountInResponse {}
+message MsgSwapExactAmountInResponse {
+           string                   tokenOutAmount = 1 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", (gogoproto.moretags) = "yaml:\"taken_out_max_amount\"", (gogoproto.nullable) = false];
+}
 
 message MsgSwapExactAmountOut {
-           string                   creator           = 1;
-           cosmos.base.v1beta1.Coin tokenOut          = 2 [(gogoproto.nullable) = false];
-           string                   tokenOutMaxAmount = 3 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Uint", (gogoproto.moretags) = "yaml:\"taken_out_max_amount\"", (gogoproto.nullable) = false];
-  repeated uint64                   swapRoutePoolIds  = 4;
-  repeated string                   swapRouteDenoms   = 5;
+           string                   sender           = 1;
+           repeated SwapAmountOutRoute routes = 2 [ (gogoproto.nullable) = false ];
+           cosmos.base.v1beta1.Coin tokenOut          = 3 [(gogoproto.nullable) = false];
+           string                   tokenInMaxAmount = 4 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", (gogoproto.moretags) = "yaml:\"taken_out_max_amount\"", (gogoproto.nullable) = false];
```

Comment:
tokenInMaxAmount vs taken_out_max_amount - I recommend to remove all moretags

Reply:
all the moretags annotations were removed